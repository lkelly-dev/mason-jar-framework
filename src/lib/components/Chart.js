import React from 'react';
import { cx } from 'emotion';
import { debugAttrs, debugMode } from '../utils/debugging';

const Chart = props => {
  const { context, children, className, ...rest } = props;
  const debug = debugMode(context)
  const chartClass = cx('chart', className);
  return (
    <div className={chartClass}
      {...(debug && debugAttrs(props)) }
      {...rest}>
      {children}
    </div>
  );
};

const ChartCanvas = props => {
  const { context, className, ...rest } = props;
  const debug = debugMode(context)
  const chartClass = cx('chart-canvas', className);
  return (
    <canvas className={chartClass}
      {...(debug && debugAttrs(props)) }
      {...rest} />
  );
};

Chart.Canvas = ChartCanvas;

export { Chart, ChartCanvas };
