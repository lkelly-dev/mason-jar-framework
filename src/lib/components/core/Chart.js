import React from 'react';
import { cx } from 'emotion';

const Chart = props => {
  const { children, className, ...rest } = props;
  const chartClass = cx('chart', className);
  return (
    <div className={chartClass} {...rest}>
      {children}
    </div>
  );
};

const ChartCanvas = props => {
  const { className, ...rest } = props;
  const chartClass = cx('chart-canvas', className);
  return <chart className={chartClass} {...rest} />;
};

Chart.Canvas = ChartCanvas;

export { Chart, ChartCanvas };
