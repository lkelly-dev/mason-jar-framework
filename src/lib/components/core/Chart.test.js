import React from 'react';
import { shallow } from 'enzyme';
import { Chart, ChartCanvas } from "./Chart";

describe('Chart', () => {
  it('Chart matches snapshot', () => {
    const component = shallow(<Chart></Chart>);
    expect(component).toMatchSnapshot();
  });

  it('has expected props', () => {
    const childComponent = <div>Child Component</div>;
    const component = shallow(
      <Chart
        className="nifty"
        what="ever" >
        { childComponent }
      </Chart>
    );
    const classNameArray = component.prop('className').split(' ');
    expect(component.contains(childComponent)).toEqual(true);
    expect(component.prop('what')).toBe('ever');
    expect(classNameArray).toContain('chart');
    expect(classNameArray).toContain('nifty');
    expect(classNameArray).toHaveLength(2);
  });
});

describe('ChartCanvas', () => {
  it('ChartCanvas matches snapshot', () => {
    const component = shallow(<ChartCanvas></ChartCanvas>);
    expect(component).toMatchSnapshot();
  });

  it('has expected props', () => {
    const childComponent = <div>Child Component</div>;
    const component = shallow(
      <ChartCanvas
        className="nifty"
        what="ever" >
        { childComponent }
      </ChartCanvas>
    );
    const classNameArray = component.prop('className').split(' ');
    expect(component.type()).toBe('canvas');
    expect(component.contains(childComponent)).toEqual(true);
    expect(component.prop('what')).toBe('ever');
    expect(classNameArray).toContain('chart-canvas');
    expect(classNameArray).toContain('nifty');
    expect(classNameArray).toHaveLength(2);
  });
});
