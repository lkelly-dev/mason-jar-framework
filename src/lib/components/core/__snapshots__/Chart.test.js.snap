// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart Chart matches snapshot 1`] = `
<div
  className="chart"
/>
`;

exports[`ChartCanvas ChartCanvas matches snapshot 1`] = `
<canvas
  className="chart-canvas"
/>
`;
